lib/services/notification-api-service.js

```diff
@@ -109,6 +109,147 @@ function NotificationApiServiceFactory(
             return message;
         });
     };
+    NotificationApiService.prototype.redfishAlertProcessing = function(req) {
+        /*
+        The alert would look like the json below when published on the AMQP bus.
+        It is compliant with RackHD notification standard.
+        Notification Type is "node" and action is "alerts"
+
+         {
+         "type": "node",
+         "action": "alerts",
+         "typeId": "58d94cec316779d4126be134",
+         "data": {
+             "Context": "context string",
+             "EventId": "8689",
+             "EventTimestamp": "2017-03-29T15:39:34-0500",
+             "EventType": "Alert",
+             "MemberId": "7e675c8e-127a-11e7-9fc8-64006ac35232",
+             "Message": "The coin cell battery in CMC 1 is not working.",
+             "MessageArgs": ["1"],
+             "MessageArgs@odata.count": 1,
+             "MessageId": "CMC8572",
+             "Severity": "Critical",
+             "sourceIpAddress": "10.240.19.130",
+             "nodeId": "58d94cec316779d4126be134",
+             "ChassisName": "PowerEdge R630",
+             "ServiceTag": "4666482",
+             "SN": "CN747515A80855"
+         },
+         "severity": "critical",
+         "version": "1.0",
+         "createdAt": "2017-03-29T19:44:12.972Z"
+         }
+         */
+        var amqpMessage = {};
+
+        return new Promise.resolve()
+            .then(function(){
+                return getAlert(req);
+            })
+            .then(function(alert){
+                return  updateBmcInfo(alert);
+            })
+            .then(function(data){
+                return updateDeviceInfo(data);
+            })
+            .then(function(data){
+                amqpMessage = data;
+                return eventsProtocol.publishExternalEvent(data)
+                    .then(function () {
+                        return amqpMessage;
+                    });
+            });
+    };
+
+    function getAlert (req) {
+        var amqpMessage = {};
+        var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress ||
+            req.socket.remoteAddress || req.connection.socket.remoteAddress;
```

Comment:
I know this code will work in your system, but in the long run, this kind of code divergence will bring problems, and something even hard to debug.

For example, this function (https://github.com/RackHD/on-http/blob/master/lib/services/http-service.js#L25) will return `X-Real-IP` as source, this is part of API proxy feature, but your code code doesn't use `X-Real-IP`, so as the API proxy feature will not touch this particular API.